File: Python/baekjoon/단계별로 풀어보기/기본 수학2/2581_소수.py
# https://www.acmicpc.net/problem/2581

def prime(n):
    import math
    if n < 2:
        return False
    elif n == 2:
        return True
    else:
        for i in range(2, math.ceil(math.sqrt(n)) + 1):
            if n % i == 0:
                return False
        return True


m = int(input())
n = int(input())
answer = []
for i in range(m, n + 1):
    if prime(i):
        answer.append(i)
if answer:
    print(sum(answer))
    print(answer[0])
else:
    print(-1)

File: C#/Baekjoon/약수, 배수와 소수/2581_소수.cs
//https://www.acmicpc.net/problem/2581

bool[] p = new bool[10001];
p[0] = true;
p[1] = true;
for (int i = 2; i < 10001; i++)
{
    if (p[i] == false)
    {
        for (int j = i; j * i < 10001; j++)
        {
            p[j * i] = true;
        }
    }
}

int n = int.Parse(Console.ReadLine());
int m = int.Parse(Console.ReadLine());
int sum = 0, min = 0;
for (int i = m; i >= n; i--)
{
    if (p[i] == false)
    {
        min = i;
        sum += i;
    }
}
Console.Write(min != 0 ? $"{sum}\n{min}" : -1);